app/scripts/modules/google/src/interceptors/iap.interceptor.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  IHttpInterceptor,
+  IHttpPromiseCallbackArg,
+  IHttpProvider,
+  IPromise,
+  IQService,
+  IRequestConfig,
+  module,
+} from 'angular';
+import { $http } from 'ngimport';
+import { SETTINGS } from '@spinnaker/core';
+import { Dictionary } from 'lodash';
+
+/**
+ * When a Cloud IAP (if enabled) session expires, refreshes session.
+ */
+
+export class IapInterceptor implements IHttpInterceptor {
+  private MAX_RETRIES = 4;
+  private retryQueue: Dictionary<number> = {};
+
+  constructor(private $q: IQService) {
+    'ngInject';
+  }
+
+  private removeFromQueue(config: IRequestConfig): void {
+    delete this.retryQueue[config.url];
+  }
+
+  public responseError = <T>(response: IHttpPromiseCallbackArg<T>): IPromise<T> => {
+    const { config, status } = response;
+    const retryCount = this.retryQueue[config.url] || 0;
+
+    if (status === 401 && retryCount < this.MAX_RETRIES && SETTINGS.feature.iapRefresherEnabled) {
```

Comment:
What's the retry logic doing here? Is it possible that the IAP flow could fail?

Reply:
Removed